### ISSUE TITLE
Incorrect suggestions for `significant_drop_in_scrutinee`

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang/rust-clippy#13986]
### Summary

The `significant_drop_in_scrutinee` lint gives suggestions that are only applicable when the expression being linted is a statement, not part as a larger expression.

### Reproducer

I tried this code:

```rust
#![warn(clippy::significant_drop_in_scrutinee)]

#[clippy::has_significant_drop]
struct S;

fn main() {
    #[allow(clippy::match_single_binding)]
    let _ = match (S, S) {
        (S, S) => 0,
    };
}
```

I expected to see this happen:

A suggestion saying to use:

```rust
   let value = (S, S);
   let _ = match value { … }
```

Instead, this happened:

```
1  | #![warn(clippy::significant_drop_in_scrutinee)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: try moving the temporary above the match
   |
8  ~     let _ = let value = (S, S);
9  ~     match value {
```

### Version

```text
rustc 1.86.0-nightly (b1a7dfb91 2025-01-10)
binary: rustc
commit-hash: b1a7dfb91106018f47ed9dc9b27aee1977682868
commit-date: 2025-01-10
host: x86_64-unknown-linux-gnu
release: 1.86.0-nightly
LLVM version: 19.1.6
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
